PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureRunProfileState.java

```diff
@@ -52,72 +50,54 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
                 }
             }).subscribeOn(SchedulerProviderFactory.getInstance().getSchedulerProvider().io()).subscribe(
                 (res) -> {
-                    if (operation != null) {
-                        operation.complete();
-                    }
-                    this.sendTelemetry(telemetryMap, true, null);
+                    this.sendTelemetry(operation, telemetryMap, null);
                     this.onSuccess(res, processHandler);
                 },
                 (err) -> {
                     err.printStackTrace();
-                    if (operation != null) {
-                        EventUtil.logError(operation, ErrorType.userError, new Exception(err.getMessage(), err),
-                                           telemetryMap, null);
-                        operation.complete();
-                    }
+                    this.sendTelemetry(operation, telemetryMap, err);
                     this.onFail(err, processHandler);
-                    this.sendTelemetry(telemetryMap, false, err.getMessage());
                 });
         return new DefaultExecutionResult(consoleView, processHandler);
     }
 
-    protected Operation createOperation() {
-        return null;
-    }
-
-    protected void onFail(@NotNull Throwable error, @NotNull RunProcessHandler processHandler) {
-        final String errorMessage = (error instanceof AzureToolkitRuntimeException || error instanceof AzureToolkitException) ?
-                                    String.format("Failed to %s", error.getMessage()) : error.getMessage();
-        processHandler.println(errorMessage, ProcessOutputTypes.STDERR);
-        processHandler.notifyComplete();
-        IntelliJAzureExceptionHandler.getInstance().handleException(project, new AzureToolkitRuntimeException("execute run configuration", error), true);
-    }
-
     protected void setText(RunProcessHandler runProcessHandler, String text) {
         if (runProcessHandler.isProcessRunning()) {
             runProcessHandler.setText(text);
         }
     }
 
-    protected void updateTelemetryMap(@NotNull Map<String, String> telemetryMap) {
-    }
-
-    private void sendTelemetry(@NotNull Map<String, String> telemetryMap, boolean success, @Nullable String errorMsg) {
+    private void sendTelemetry(Operation operation, @NotNull Map<String, String> telemetryMap, Throwable exception) {
         updateTelemetryMap(telemetryMap);
-        telemetryMap.put("Success", String.valueOf(success));
-        if (!success) {
-            telemetryMap.put("ErrorMsg", errorMsg);
+        operation.trackProperties(telemetryMap);
+        if (exception != null) {
+            EventUtil.logError(operation, ErrorType.userError, new Exception(exception.getMessage(), exception), telemetryMap, null);
         }
-
-        AppInsightsClient.createByType(AppInsightsClient.EventType.Action
-                , getDeployTarget(), "Deploy", telemetryMap);
+        operation.complete();
```

Comment:
why delete `AppInsightsClient.createByType(AppInsightsClient.EventType.Action
                , getDeployTarget(), "Deploy", telemetryMap);`

Reply:
Its the outdated telemetry( not start with `AzurePlugin.Intellij/*`), which is not used now.